src/storage/v2/durability/snapshot.cpp

```diff
@@ -225,20 +226,27 @@ void LoadPartialEdges(const std::filesystem::path &path, utils::SkipList<Edge> &
   if (!snapshot.SetPosition(from_offset)) throw RecoveryFailure("Couldn't read data from snapshot!");
 
   std::vector<std::pair<PropertyId, PropertyValue>> read_properties;
+  uint64_t five_percent_chunk = edges_count / 20;
+  uint64_t percentage_delta = 0;
+  spdlog::debug("Started to recover edge set from 0 to {} vertices.", 0 + five_percent_chunk);
   for (uint64_t i = 0; i < edges_count; ++i) {
+    if (i > 0 && i % five_percent_chunk == 0) {
+      percentage_delta += 5;
+      spdlog::info("Recovered {}% of edges.", percentage_delta);
+      if (percentage_delta != 100) spdlog::debug("Started to recover edge set <{} - {}>", i, i + five_percent_chunk);
```

Comment:
IMO minor edge case for displaying the last debug
`(percentage_delta != 95) ? (i + five_percent_chunk) : (edges_count - 1)`

Reply:
we'll keep it as is for now

Reply:
This has been included in latest commit.